### PR TITLE
Fix hidden variables

### PR DESCRIPTION
**Pull request by chowington**
Fixes https://github.com/VEuPathDB/web-eda/issues/1330

### CODE REVIEW

**Review by dmfalke — CHANGES_REQUESTED** [review]
This looks good! Could you add a short comment explaining what `rootParentIds` is, and how we decide what gets added to that array? Thanks!

File: src/lib/core/utils/wdk-filter-param-adapter.ts [hunk 1]
```diff
@@ -286,8 +282,20 @@ export function makeHiddenVariablesInScope(
     });
   }
 
-  variablesByParentId[entity.id]?.forEach((variable) => {
-    _traverseDescendantVariables(variable, false);
+  const variableIds = new Set(entity.variables.map((variable) => variable.id));
```
Comment (chowington) [inline src/lib/core/utils/wdk-filter-param-adapter.ts:285]: Note that this is duplicated in `entitiesToFields`. Is it worth deduplicating this operation?

Comment (dmfalke) [reply src/lib/core/utils/wdk-filter-param-adapter.ts:285]: I think we should, eventually; but I think we will want to overhaul how we are handling variable trees in a larger context. You'll notice that we are doing a pseudo tree traversal of variables here, which means we have to re-implement the "parent logic" here.

An overhaul I've had in mind (but haven't had time to do, yet) is to construct the tree in the subsetting client, and to make that available as a part of the study metadata that the application code uses. I don't think there is an issue for this. I will make one in a bit and reference it here.

Comment (dmfalke) [reply src/lib/core/utils/wdk-filter-param-adapter.ts:285]: https://github.com/VEuPathDB/web-eda/issues/1434

**Review by dmfalke — APPROVED** [review]

**Pull request merged**